Panic `cap out of range` on query

Just hit the following while running a query from the UI:
```
panic: runtime error: makeslice: cap out of range

goroutine 159202 [running]:
github.com/parca-dev/parca/pkg/query.queryToFilterExprs({0xc00c33e9a8?, 0xc03c3c93b0?})
	github.com/parca-dev/parca/pkg/query/columnquery.go:193 +0x5f
github.com/parca-dev/parca/pkg/query.(*ColumnQueryAPI).QueryRange(0xc0011cc370, {0x359def2?, 0x2c2f060?}, 0xc01d76a3c0)
	github.com/parca-dev/parca/pkg/query/columnquery.go:244 +0x165
github.com/parca-dev/parca/gen/proto/go/parca/query/v1alpha1._QueryService_QueryRange_Handler.func1({0x3f8d928, 0xc0152d41e0}, {0x3382a20?, 0xc01d76a3c0})
	github.com/parca-dev/parca/gen/proto/go/parca/query/v1alpha1/query_vtproto.pb.go:181 +0x78
github.com/grpc-ecosystem/go-grpc-middleware/v2/interceptors.UnaryServerInterceptor.func1({0x3f8d928, 0xc0152d41e0}, {0x3382a20, 0xc01d76a3c0}, 0x0?, 0xc015144dc8)
	github.com/grpc-ecosystem/go-grpc-middleware/v2@v2.0.0-rc.2.0.20201207153454-9f6bf00c00a7/interceptors/server.go:22 +0x21e
github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1({0x3f8d928?, 0xc0152d41e0?}, {0x3382a20?, 0xc01d76a3c0?})
	github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:25 +0x3a
github.com/grpc-ecosystem/go-grpc-prometheus.(*ServerMetrics).UnaryServerInterceptor.func1({0x3f8d928, 0xc0152d41e0}, {0x3382a20, 0xc01d76a3c0}, 0x0?, 0xc00de6b2c0)
	github.com/grpc-ecosystem/go-grpc-prometheus@v1.2.0/server_metrics.go:107 +0x87
github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1({0x3f8d928?, 0xc0152d41e0?}, {0x3382a20?, 0xc01d76a3c0?})
	github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:25 +0x3a
go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc.UnaryServerInterceptor.func1({0x3f8d928, 0xc03befff50}, {0x3382a20, 0xc01d76a3c0}, 0xc00de6b2a0, 0xc00de6b300)
	go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc@v0.32.0/interceptor.go:325 +0x664
github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1({0x3f8d928?, 0xc03befff50?}, {0x3382a20?, 0xc01d76a3c0?})
	github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:25 +0x3a
github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1({0x3f8d928, 0xc03befff50}, {0x3382a20, 0xc01d76a3c0}, 0xc020eecaf0?, 0x2ec5320?)
	github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:34 +0xbf
github.com/parca-dev/parca/gen/proto/go/parca/query/v1alpha1._QueryService_QueryRange_Handler({0x3382b60?, 0xc0011cc370}, {0x3f8d928, 0xc03befff50}, 0xc01d76e420, 0xc000fd81e0)
	github.com/parca-dev/parca/gen/proto/go/parca/query/v1alpha1/query_vtproto.pb.go:183 +0x138
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000ff8000, {0x3f9c050, 0xc0011ec990}, 0xc01d23e000, 0xc000fd84b0, 0x5a468c0, 0x0)
	google.golang.org/grpc@v1.46.0/server.go:1283 +0xcfd
google.golang.org/grpc.(*Server).handleStream(0xc000ff8000, {0x3f9c050, 0xc0011ec990}, 0xc01d23e000, 0x0)
	google.golang.org/grpc@v1.46.0/server.go:1620 +0xa1b
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	google.golang.org/grpc@v1.46.0/server.go:922 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.46.0/server.go:920 +0x28a
```

Parca version: v0.11.1

--- a/pkg/query/columnquery.go
+++ b/pkg/query/columnquery.go
@@ -195,10 +195,10 @@ func queryToFilterExprs(query string) ([]logicalplan.Expr, error) {
 		return nil, status.Error(codes.InvalidArgument, "failed to parse query")
 	}
 
-	sel := make([]*labels.Matcher, 0, len(parsedSelector)-1)
+	sel := make([]*labels.Matcher, 0, len(parsedSelector))
 	var nameLabel *labels.Matcher
 	for _, matcher := range parsedSelector {
-		if matcher.Name == "__name__" {
+		if matcher.Name == labels.MetricName {
 			nameLabel = matcher
 		} else {
 			sel = append(sel, matcher)

